Tutorial company level experiments

Lior539 opened the pull request:
## Changes

Looking for feedback on:
* Is it interesting? 
* Should I cover anything else
* What should I remove 
* Line edits
* I'm not sure of the phrases “group-level” or “user-level”. Is this clear? Is there a better term?

## Checklist
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Feature names are in **[sentence case too]([https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case))**
- [x] Words are spelled using American English
- [x] I have checked out our [style guide](https://github.com/PostHog/posthog.com/blob/master/STYLEGUIDE.md)
- [x] If I moved a page, I added a redirect in `vercel.json`

#5798

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+sidebar: Docs
+showTitle: true
+author: ['lior-neu-ner']
+date: // what to put here?
```
I usually put the current date to start (so it shows up at the top locally) then change before publishing.

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+## When to run group-level experiments
+
+When the change in how an individual uses your product will greatly affect the behavior of other users, or when you want to measure the impact of a change on an entire group of users. This is usually seen in the following types of products:
```
Try to be consistent with what you name one person, so far you use "person", "individual", and "user" when you should probably just use one of these

Lior539 replied:
That's a good point. I'll keep it to user as much as possible

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+3. Marketplace products
+   
+   - For example, the Uber example mentioned previously.
```
Could you add another example here, feels a bit weak to say this.

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+Randomization risk can occur when companies or groups are not properly randomized, which can distort results.
+
+For example, consider a B2B SaaS company that wants to test the impact of a new pricing plan on customer acquisition and retention. They decide to run a group-level experiment, randomly assigning companies to either the new pricing plan (test group) or the existing pricing plan (control group). However, suppose the randomization process inadvertently assigns a disproportionate number of small com ...
```
This is really long, it should be like 1/2 the size

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+To improve the quality of insights from group-level changes and address the risks mentioned above, you may want to consider running an experiment with your change at a user-level before running it at a group-level. This can help you in the following ways:
+
+  ### 1. Save time for changes that don't have an impact. 
```
```suggestion
### 1. Save time for changes that don't have an impact. 
```

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+This can also help you identify randomization risks. For example, let's say a B2B Saas app runs a user-level experiment and notices that users in large companies are most likely to interact with their new feature. In this case, when they run a group-level experiment, they can ensure to filter out small and medium-sized companies in their experiment in order to see more accurate results.
+
+
+## How to setup group-level tests on PostHog
```
```suggestion
## How to setup group-level experiments on PostHog
```
Make sure to use the right names for things

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+   - For example, suppose Asana wants to test a new AI feature that automatically assigns tasks based on their urgency and importance. By conducting a group-level experiment, they can measure the impact on project completion rates across the entire company. 
+
+2. Products with network effects.
```
Is this relevant to products with individual network effects? Or just group network effects?

Lior539 replied:
Hmm Im not sure I understand if there would be a difference? Can you elaborate on what you mean?

ivanagas replied:
I was thinking something, but I think what you have as right. 

For example, Facebook has individual-level network effects. New individuals benefit existing individuals. Group network effects is like new organizations benefiting existing organizations. For example, if a new organization joins Slack that benefits existing organizations who want to Slack Connect with them. 

Again, I think I'm wrong and you're right 😄

ivanagas commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+
+1. B2B SaaS apps
+
+   - For example, suppose Asana wants to test a new AI feature that automatically assigns tasks based on their urgency and importance. By conducting a group-level experiment, they can measure the impact on project completion rates across the entire company. 
```
Remove indents

Lior539 replied:
the indents are intentional. I'd like the bullets to be nested within the numbered list (I'll check that it's actually doing this on the site)

ivanagas replied:
Yup, ran it locally and works.

ivanagas reviewed (commented):
A good start. Some higher level thoughts:
1. Make sure to take a read through this: https://posthog.com/handbook/growth/marketing/posthog-style-guide
2. I think the order should be: When, How, Reduce Risk.

Lior539 commented on contents/tutorials/how-to-run-successful-group-experiments.md:
```diff
@@ -0,0 +1,86 @@
+   
+   - For example, the Uber example mentioned previously.
+
+## How does running a group-level experiment differ from a user-level one?
```
> We should follow PostHog terminology and use "person-level" I think

I thought about it, but user-level is more common in the industry (AFAIK posthog is the only place Ive come across "Person")

Lior539 commented:
Changes:
* Changed "group-level" and "user-level" to group-targeted and user-targeted (in most places)
* Changed title to "When and how to run group-targeted A/B tests" as its more accurate description of whats discussed in the post
* add other sections, more text

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+---
+title: When and how to run group-targeted A/B tests
```
This is a good title, but you should tighten up the slug.

- running-group-targeted-ab-tests
- group-targeted-ab-tests

The "how and when to run" part isn't super relevant for the the slug.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+tags: ['experimentation']
+---
+
+A/B tests are a powerful tool for measuring how product changes impact user behavior. However, sometimes changing how one user interacts with your product will affect how others use it too. For example, a change in the Uber App for Drivers will likely affect riders' experience too. In these scenarios, group-targeted experiments enable us to measure the impact of changes beyond an individual user  ...
```
```suggestion
A/B tests are a powerful tool for measuring how product changes impact user behavior. However, sometimes changing how one user interacts with your product will affect how others use it too. For example, a change in the Uber app for drivers will likely affect riders' experience too. In these scenarios, group-targeted experiments enable us to measure the impact of changes beyond an individual user and across a group of users.
```

ivanagas replied:
```suggestion
A/B tests are a powerful tool for measuring how product changes impact user behavior. However, sometimes changing how one user interacts with your product will affect how others use it too. For example, a change in the Uber app for drivers likely affects riders' experience too. In these scenarios, group-targeted experiments enable us to measure the impact of changes beyond an individual user and across a group of users.
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+## What are group-targeted experiments
+
+In group-targeted experiments, the changes being tested are applied to an entire group, rather than individual users. This means that everyone within a specific group will experience the same conditions. For example, either the control or test variant. This enables you to measure the impact of any changes on the group as a whole, rather than on individual users within the group.
```
```suggestion
In group-targeted experiments, the test applies to an entire group, rather than individual users. This means that everyone within a specific group will experience the same conditions, for example, either the control or test variant. This enables you to measure the impact of any changes on the group as a whole, rather than on individual users within the group.
```
1. Make sure to use [active voice](https://writing.wisc.edu/handbook/style/ccs_activevoice/)
2. "This enables you to measure the impact of any changes on the group as a whole, rather than on individual users within the group." is very similar to "In these scenarios, group-targeted experiments enable us to measure the impact of changes beyond an individual user and across a group of users." from the previous. You could probably cut or change one of them.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+## When to run group-targeted experiments
+
+When the change in how an individual user uses your product will significantly affect the behavior of other users, or when you want to measure the impact of a change on an entire group of users. This is usually seen in the following types of products:
```
```suggestion
Run group-targeted experiments when:
1. The change in how an individual user uses your product significantly affects the behavior of other users.
2. You want to measure the impact of a change on an entire group of users. 
This is usually seen in the following types of products:
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+When the change in how an individual user uses your product will significantly affect the behavior of other users, or when you want to measure the impact of a change on an entire group of users. This is usually seen in the following types of products:
+
+1. B2B SaaS apps, in order to test how a change will affect how an entire company uses your product.
```
```suggestion
1. B2B SaaS apps, to test how a change affects how an entire company uses your product.
```

ivanagas replied:
You kind of repeat yourself here too. You could just combine this with this with the introduction. Add the example to the bullet in the first paragraph of this sentence

> 2. You want to measure the impact of a change on an entire group of users.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+   - For example, suppose Asana wants to test a new AI feature that automatically assigns tasks based on their urgency and importance. By conducting a group-targeted experiment, they can measure the impact on project completion rates across the entire company. 
+
+2. Products with network effects, since a change for one user will likely affect how other users interact with the product.
```
Same

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+   - For example, suppose Slack wants to improve the usage of a new video calling feature. Improving the feature's discoverability for a single user will increase their own usage with it, but since they use it with their coworkers, their coworkers will also discover it.
+
+## How does running a group-targeted experiment differ from a user-targeted one?
```
This section should be after the how-to section

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+Randomization risk can occur when groups are not properly randomized when they are assigned to the control or test variants. Since in group-targeted experiments you have fewer data points, the risk is higher that this will occur and this can distort results.
+
+A B2B SaaS company wants to see if a new pricing plan helps retain customers. They split companies into two groups: one with the new plan and one with the old plan. However, the randomization process assigns more small companies to the new plan and more large companies to the old plan. This may skew the results because company size could be a factor in determining customer retention rather than t ...
```
```suggestion
For example, a B2B SaaS company wants to see if a new pricing plan helps retain customers. They split companies into two groups: one with the new plan and one with the old plan. However, the randomization process assigns more small companies to the new plan and more large companies to the old plan. This may skew the results because company size could be a factor in determining customer retention rather than the pricing plan itself.
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+Since group-targeted experiments provide insights on an aggregate level, they may not show insights on a user level.
+
+For example, say a B2B SaaS tool wants users to invite coworkers to use the app. They add a big "Invite Coworker" button and test it in a company-targeted experiment. The results show more invites are sent and new users join, but it's hard to know how individual users used the button. For example, what are the roles or skill sets of team members who are most likely to interact with the button and ...
```
Why can't you just create an insight for individual? It doesn't seem like group-target experiments prevent you from getting data from individuals.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+To improve the quality of insights from group-targeted experiments and address the risks mentioned above, you may want to consider running an experiment with your change at a user-level before running it at a group-level. This can help you in the following ways:
+
+### 1. Save time for changes that don't have an impact. 
```
```suggestion
### 1. Run user-targeted experiments first
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+This can also help you identify randomization risks. For example, let's say a B2B SaaS app runs a user-targeted experiment and notices that users in large companies are more likely to interact with their new feature. In this case, when they run a group-targeted experiment, they can ensure to filter out small and medium-sized companies in their experiment in order to obtain more accurate results.
+
+
+## How to setup group-targeted experiments on PostHog
```
If this is a tutorial, then move this up.

If this is going to be a blog, then you can just remove this and rename the piece "when and why to run group targeted a/b tests" (or something like that.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+As mentioned, lower statistical power in group-targeted experiments means you usually need to run your experiments longer. However, a change that does not impact behavior at a user-level will not translate into impact at a group-level. So you may waste time waiting for results from a group-level test that has no impact.
+
+In this case, it may be worth running a user-targeted experiment first in order to see if there is any impact on individual user behavior (which should be quicker to gain insights from because of higher statistical power) before running a group-targeted experiment.
+
```
```suggestion
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+### 1. Save time for changes that don't have an impact. 
+
+As mentioned, lower statistical power in group-targeted experiments means you usually need to run your experiments longer. However, a change that does not impact behavior at a user-level will not translate into impact at a group-level. So you may waste time waiting for results from a group-level test that has no impact.
```
```suggestion
As mentioned, lower statistical power in group-targeted experiments means you need to run your experiments longer. However, a change that does not impact behavior at a user-level likely will not translate into impact at a group-level. So you may waste time waiting for results from a group-level test that has no impact.
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+By running a user-targeted experiment first, you'll gain insight into how individual users interact with your change and what characteristics they may possess. This will then give you an understanding of what results you can expect from a group-targeted experiment.
+
+This can also help you identify randomization risks. For example, let's say a B2B SaaS app runs a user-targeted experiment and notices that users in large companies are more likely to interact with their new feature. In this case, when they run a group-targeted experiment, they can ensure to filter out small and medium-sized companies in their experiment in order to obtain more accurate results.
+
```
```suggestion
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+
+First, ensure you have [setup a group](docs/product-analytics/group-analytics#getting-started) in PostHog. Next, go to your PostHog dashboard and create a [new experiment](docs/experiments/manual#creating-an-experiment).
+
+In the New Experiment screen, you'll be able to select the Participant type. By default, it will show `Persons` (i.e. a user-level experiment). Click on the drop-down and select your new group.
```
```suggestion
In the New Experiment screen, select the Participant type. By default, it will show `Persons` (i.e. a user-level experiment). Click on the drop-down and select your group.
```

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+Then set your experiment goal and secondary metrics as usual. A handy feature that's especially useful for group-level experiments is the Minimum Acceptable Improvement calculator at the bottom of the page. Since group-level experiments have lower statistical power, this will recommend a time for you for how long you should run your experiment in order to see results.
+
+![Screenshot of the Minimum Acceptable Improvement calculator](../images/tutorials/when-and-how-to-run-group-targeted-a-b-tests/minimum-acceptable-improvement.png)
+
```
There are some more steps missing here:
1. Click create experiment
2. Implement, add flags to code
3. Launch, wait for results
4. How to understand if your experiment is relevant, had an impact, and avoided risks (this ties this section to the rest of the piece)

Lior539 replied:
Hmm Im not sure we need to go into that much detail. The specifics of how to create and run an experiment as usual are hyperlinked above

ivanagas reviewed (commented):
Better, but still some changes to be made.

1. Decide whether this will be blog or tutorial, could go both ways.
2. Use active voice, try not to repeat yourself.

ivanagas commented on contents/tutorials/when-and-how-to-run-group-targeted-a-b-tests.md:
```diff
@@ -0,0 +1,88 @@
+showTitle: true
+author: ['lior-neu-ner']
+date: 2023-04-21
+featuredImage: ../images/tutorials/when-and-how-to-run-group-targeted-a-b-tests/user-vs-group-a-b-tests.png
```
This unfortunately looks weird as preview image
![Screen Shot 2023-04-21 at 9 44 38 AM](https://user-images.githubusercontent.com/34755028/233690735-14e1eaa5-3589-4e55-989a-6001dbea57a3.png)

ivanagas commented on contents/blog/running-group-targeted-ab-tests.md:
```diff
@@ -0,0 +1,93 @@
+
+Since you treat a group of users as a single data point in group-targeted experiments, you have less statistical power. For example, Slack has 20 million users, but only 600,000 companies using it.
+
+In practice, this means that you'll usually have to run group-targeted experiments longer than you would user-targeted experiments. It also means that your results may be more susceptible to noise.
```
"More susceptible to noise" is a little unclear in this context, could use with some more clarification. Maybe another sentence on why noise is bad? It makes it less clear the success of an experiment?

ivanagas commented on contents/blog/running-group-targeted-ab-tests.md:
```diff
@@ -0,0 +1,93 @@
+
+As mentioned, lower statistical power in group-targeted experiments means you need to run your experiments longer. However, a change that does not impact behavior at a user-level will not likely translate into impact at a group-level. So you may waste time waiting for results from a group-level test that has no impact.
+
+In this case, it may be worth running a user-targeted experiment first in order to see if there is any impact on individual user behavior (which should be quicker to gain insights from because of higher statistical power) before running a group-targeted experiment.
```
```suggestion
In this case, it may be worth running a user-targeted experiment first in order to see if there is any impact on individual user behavior (where it should be quicker to gain results from because of higher statistical power) before running a group-targeted experiment.
```

ivanagas commented on contents/blog/running-group-targeted-ab-tests.md:
```diff
@@ -0,0 +1,93 @@
+showTitle: true
+author: ['lior-neu-ner']
+date: 2023-04-24
+featuredImage: ../images/blog/running-group-targeted-ab-tests/featured-image.png
```
White might be a little jarring as a background colour, maybe try the same color as the tutorial backgrounds? (E5E7E0)

Lior539 replied:
think I'll just ask Lottie for an image. Fits in better with the other blogs as well

andyvan-ph replied:
Assuming this stays in the tutorials section, it should retain the same style. I'm sure there's one we can reuse.

andyvan-ph replied:
Ah, I see it's moved to the blog already. That being the case, @Lior539 you need to update the frontmatter to the blog version. You can grab it from any existing blog post.

ivanagas reviewed (approved):
Looks good, a few last minor comments.

andyvan-ph commented on contents/blog/running-group-targeted-ab-tests.md:
```diff
@@ -0,0 +1,106 @@
+
+### 2. Gain user-level insights and identify randomization risks
+
+By running a user-targeted experiment first, you'll gain insight into how individual users interact with your change and what characteristics they may possess. This will then give you an understanding of what results you can expect from a group-targeted experiment.
+
+This can also help you identify randomization risks. For example, let's say a B2B SaaS app runs a user-targeted experiment and notices that users in large companies are more likely to interact with their new feature. In this case, when they run a group-targeted experiment, they can ensure to filter out small and medium-sized companies in their experiment in order to obtain more accurate results.
```
```suggestion
Running a user-targeted experiment first gives you insight into how individual users interact with your change and what properties they possess. This will then give you an understanding of what results you can expect from a group-targeted experiment.

This can also help you identify randomization risks. For example, let's say a B2B SaaS app runs a user-targeted experiment, and notices that users in large companies are more likely to interact with their new feature. In this case, when they run a group-targeted experiment, they can ensure to filter out small and medium-sized companies in their experiment in order to obtain more accurate results.
```

andyvan-ph commented on contents/blog/running-group-targeted-ab-tests.md:
```diff
@@ -0,0 +1,106 @@
+
+![Screenshot of the Minimum Acceptable Improvement calculator](../images/blog/running-group-targeted-ab-tests/minimum-acceptable-improvement.png)
+
+
```
This article needs a summary to wrap it up – basically imagine a TL;DR. Something like:

"To summarize, remember these x points when running group-targeted experiments:

1. XYZ
2. XYZ
3. XYZ

To learn more about running product experiments, read our guides to `links`.

andyvan-ph reviewed (changes requested):
@Lior539 This very close, but just a few line edits and a request to add a summary section at the end.

Lior539 commented:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FkE3pv9YYpMZVugjPwQ%2Fgiphy-downsized-medium.gif&token=ohyeah" data-gitmeme-token="ohyeah"><img src="https://media1.giphy.com/media/kE3pv9YYpMZVugjPwQ/giphy-downsized-medium.gif" title="Created by gitme.me with /ohyeah"/></a>
